File: solutions/haiku-1/haiku/haskell/transpose.hs
import Data.List (transpose)

main :: IO ()
main = do
  line1 <- getLine
  let [r, c] = map read (words line1) :: [Int]
  matrix <- sequence [getLine | _ <- [1..r]]
  let parsedMatrix = map (map read . words) matrix :: [[Int]]
  let transposed = transpose parsedMatrix
  mapM_ printRow transposed
  where
    printRow row = putStrLn (unwords (map show row))

File: solutions/haiku-1/haiku/ruby/transpose.rb
# Read dimensions
r, c = gets.split.map(&:to_i)

# Read the matrix
matrix = []
r.times do
  row = gets.split.map(&:to_i)
  matrix << row
end

# Transpose and output
matrix.transpose.each do |row|
  puts row.join(' ')
end
